### ISSUE TITLE
[code-block-tools] is ignored in pyproject.toml under [tool.rumdl]

### ISSUE THREAD

**Issue by maxencefaldor** [rvben/rumdl#851]
### rumdl version

0.2.64

### Installation method

Other (`uvx rumdl`, the PyPI wheel)

### Bug description

`[code-block-tools]` is silently ignored when it lives in `pyproject.toml` under `[tool.rumdl]`. The same section works in `.rumdl.toml`.

The docs say the four config locations are ["equivalent in capability"](https://github.com/rvben/rumdl/blob/fc410f77a2167e1229eebfcc8c6ccd269e818fe5/docs/configuration/index.md#L44-L45), which is not true for this section.

Cause, as far as I can tell from the source:

- [`parse_rumdl_toml`](https://github.com/rvben/rumdl/blob/fc410f77a2167e1229eebfcc8c6ccd269e818fe5/src/config/parsers.rs#L786-L811) has a branch that reads `[code-block-tools]` and pushes it into the fragment.
- [`parse_pyproject_toml`](https://github.com/rvben/rumdl/blob/fc410f77a2167e1229eebfcc8c6ccd269e818fe5/src/config/parsers.rs#L56) has no such branch, so the key falls through to the "Unknown global option" validation warning.
- The pyproject parser does end with a [guard that marks `code_block_tools` as withheld](https://github.com/rvben/rumdl/blob/fc410f77a2167e1229eebfcc8c6ccd269e818fe5/src/config/parsers.rs#L535-L540), referring to a field nothing in that function populates, so it looks like the parse step was simply never mirrored.

### Steps to reproduce

1. Create `pyproject.toml`:

   ```toml
   [tool.rumdl]
   disable = ["MD013"]

   [tool.rumdl.code-block-tools]
   enabled = true

   [tool.rumdl.code-block-tools.languages.python]
   lint = ["ruff:check"]
   format = ["ruff:format"]
   ```

2. Create `c.md`:

   ````markdown
   # T

   ```python
   import os
   y = [1,2]
   ```
   ````

3. Run `uvx --with ruff rumdl check c.md`:

   ```text
   [config warning] Unknown global option in pyproject.toml: code-block-tools
   Success: No issues found in 1 file (5ms)
   ```

   `disable = ["MD013"]` in the same table is honored, so the file itself is being read. Nesting the section under `[tool.rumdl.global]`, spelling it `code_block_tools`, and writing it as an inline table all produce the same warning.

4. Move the same section to `.rumdl.toml` (top-level `[code-block-tools]`, no `tool.rumdl` prefix) and rerun:

   ```text
   c.md:4:1: [ruff:check] I001 Import block is un-sorted or un-formatted
   c.md:4:8: [ruff:check] F401 `os` imported but unused
   ```

### Expected behavior

`[tool.rumdl.code-block-tools]` in `pyproject.toml` behaves like `[code-block-tools]` in `.rumdl.toml`, and the pyproject parser's known-key list includes `code-block-tools` so no warning is emitted.

### Workaround

Keep all rumdl config in `.rumdl.toml`. Splitting does not help, since only the first config file found at a directory level is used.

**Comment by rvben** [comment]
Thanks @maxencefaldor, and you had the cause right: the section fell straight through to the unknown-key path, which is why you got the "Unknown global option" warning instead of the tools running.

`[tool.rumdl.code-block-tools]` in `pyproject.toml` is parsed now, with the same shape as `.rumdl.toml`. A genuinely misspelled section still warns. Fixed in v0.2.65.
